pylsp/plugins/rope_autoimport.py

```diff
@@ -0,0 +1,219 @@
+# Copyright 2022- Python Language Server Contributors.
+
+import logging
+from typing import Any, Dict, Generator, List, Set
+
+import parso
+from jedi import Script
+from parso.python import tree
+from parso.tree import NodeOrLeaf
+from rope.base.resources import Resource
+from rope.contrib.autoimport.defs import SearchResult
+from rope.contrib.autoimport.sqlite import AutoImport
+
+from pylsp import hookimpl
+from pylsp.config.config import Config
+from pylsp.workspace import Document, Workspace
+
+log = logging.getLogger(__name__)
+
+_score_pow = 5
+_score_max = 10**_score_pow
+MAX_RESULTS = 1000
+
+
+@hookimpl
+def pylsp_settings() -> Dict[str, Dict[str, Dict[str, Any]]]:
+    # Default rope_completion to disabled
+    return {"plugins": {"rope_autoimport": {"enabled": True, "memory": False}}}
+
+
+def _should_insert(expr: tree.BaseNode, word_node: tree.Leaf) -> bool:
+    """
+    Check if we should insert the word_node on the given expr.
+
+    Works for both correct and incorrect code. This is because the
+    user is often working on the code as they write it.
+    """
+    if len(expr.children) == 0:
+        return True
+    first_child = expr.children[0]
+    if isinstance(first_child, tree.EndMarker):
+        if "#" in first_child.prefix:
+            return False  # Check for single line comment
+    if first_child == word_node:
+        return True  # If the word is the first word then its fine
+    if len(expr.children) > 1:
+        if any(node.type == "operator" and "." in node.value
+               or node.type == "trailer" for node in expr.children):
```

Comment:
```suggestion
        if any(node.type == "operator" and "." in node.value or
               node.type == "trailer" for node in expr.children):
```

I think this should solve the style issue flagged by our CIs.